pkg/hwinfo/hwinfo_nonwindows.go

```diff
@@ -41,46 +43,49 @@ func fetchInventory() (map[string]interface{}, error) {
 
 	res := make(map[string]interface{})
 
+	// all below requests are based on parsed data returned by dmidecode.Unmarshal
+	// refer to doc dmidecode.Get to get description of function behavior
```

Comment:
My intention was that you explain why you check against `err == nil` instead the "normal" `err != nil`
Explain what you explained to me in Slack, that `err == nil` means the value is present and `len() > 0`...

Reply:
`dmidecode.Get` has in description
```
// When function returns nil then dst contains slice with at least one valid entry
```
Do we need to copy?

Reply:
yeah, make it as easy as possible to understand your code without the need to dig down the libraries